apps/frontend/src/libs/modules/api/base-http-api.ts

```diff
@@ -81,6 +81,18 @@ class BaseHTTPApi implements HTTPApi {
 		return headers;
 	}
 
+	private getUrl(path: string, query?: Record<string, unknown>): string {
+		if (!query) {
+			return path;
+		}
+
+		const queryParameters = new URLSearchParams(
+			query as Record<string, string>,
+		);
```

Comment:
Hm, looks like it's not allowed to pass a number, in this case the type above will be invalid. Let's use `ConstructorParameters<typeof URLSearchParams>` instead of `Record<string, unknown>`. And this case we will need to convert number query params to string (as it was before)
```suggestion
		const queryParameters = new URLSearchParams(
			query
		);
```

Reply:
It’s very strange, but if I give `query` type of `ConstructorParameters<typeof URLSearchParams>` then I get an error: 
```
Argument of type '[init?: string | Record<string, string> | string[][] | URLSearchParams | undefined]' is not assignable to parameter of type 'string | Record<string, string> | string[][] | URLSearchParams | undefined'.
  Type '[init?: string | Record<string, string> | string[][] | URLSearchParams | undefined]' is not assignable to type 'string[][]'.
    Type 'string | Record<string, string> | string[][] | URLSearchParams | undefined' is not assignable to type 'string[]'.
      Type 'undefined' is not assignable to type 'string[]'.
```
So I need to use `ConstructorParameters<typeof URLSearchParams>[0]` (also, 0 is magic number there) as a type. Maybe it is better to leave `Record<string, string>`?

Reply:
try to use ConstructorParameters<typeof URLSearchParams>[number]

Reply:
We can't leave Record<string, string> because we may have nested fields and arrays

Reply:
That works, thank you!